DependencyInjectionContainer/Service.cs

```diff
@@ -90,6 +100,76 @@ object GetCreatedImplementationForService()
 
             var createdImplementation = ctor.Invoke(parameters);
             return createdImplementation;
+
+            ConstructorInfo GetAppropriateConstructor()
```

Comment:
Avoid using local method definitions, moreover if they are linked. Consider moving code to private methods or separate classes